lib/compat/wordpress-6.1/block-editor-settings.php

```diff
@@ -81,6 +81,18 @@ function gutenberg_get_block_editor_settings( $settings ) {
 				$block_classes['css'] = $actual_css;
 				$new_global_styles[]  = $block_classes;
 			}
+		} elseif ( current_theme_supports( 'wp-block-styles' ) ) {
```

Comment:
> Without the flex rules, Row and Stack variations will stop working altogether. Is this a problem? Would we be able to just hide them for classic themes? With blocks like Buttons we should also hide the layout controls from the sidebar if we remove the flex rules.

Great question. I was a little torn on this one — I've seen feedback from folks who would like to completely opt-out of generated Layout styles altogether, so I'd wondered if tying Layout styles to `wp-block-styles` would be a convenient way to group them. But both you and the doc make an excellent point about structural styles that otherwise break things altogether. So perhaps this line should be an `else` statement, and in either this PR (or a follow-up) we could add a way for themes that wish to opt-out of structural / layout styles altogether, an explicit way to do that, instead of an implicit one?

Reply:
> I've seen feedback from folks who would like to completely opt-out of generated Layout styles altogether

Would that be because of the styles themselves, or how they're currently output? I wonder if that might change once we remove all the duplication etc..

The thing with Layout is if we make it opt-in, and eventually migrate other core blocks to use it - maybe Columns, or Gallery once we have a grid layout - then we'll end up in this weird situation where half the blocks aren't styled.

Or should we have an opt-out of block styles altogether?

I don't really have an answer here 😅 but perhaps it's best to have a discussion about what opting in to `wp-block-styles` should mean before making a decision, so leaving it for a follow-up sounds good!

Reply:
> but perhaps it's best to have a discussion about what opting in to wp-block-styles should mean before making a decision, so leaving it for a follow-up sounds good!

Agreed — I think tomorrow, I'll switch this over to a simple `else` and we can then look at "options for opting-out of Layout styles" as a separate task once this lands 👍

Reply:
I've updated this in https://github.com/WordPress/gutenberg/pull/40875/commits/78f5c48f9bb939c6e1e64b0a7e368214b3571fe1 to remove the connection to `wp-block-styles` — I've kept a couple of the variables around, so that it's easier in a follow-up to extend this with an explicit opt-out of some kind. It seems to be testing okay for me so far, but might need a little more manual testing to make sure I haven't missed anything!

Reply:
Just retested and can confirm that flex styles work as expected on classic themes that don't opt in to `wp-block-styles`.